src/types.js

```diff
@@ -129,6 +129,17 @@ export type UserIdMap = {
   [userId: string]: User,
 };
 
+export type UserStatus = {|
+  away?: boolean,
+  status_text?: string,
```

Comment:
What's the semantics of this?

In particular, how does `away: undefined`, or a missing `away`, relate to `away: false` or `away: true`?

And is there a distinction in the meanings of `away: undefined` vs. a missing `away`? (And same question for `user_status`.)

Reply:
As other API calls both for input/output we can specify one or more of these at once. Currently, there are two but maybe more of these would be added in future. So any combination of the keys existing is possible except for no keys specified.

Reply:
OK, but my question was about the semantics -- what does it *mean* for `away` to be `true`, vs. `false`, vs. `undefined`, vs. not present? 

In particular, is there a difference in meaning between `undefined` and not present, for either property? And for `away`, does `undefined` mean the same thing as `false`, or what is the contrast in their meanings?

Reply:
Also, note this is a type definition in `src/types.js`, which we use for our own internal data structures. What the same-named parameters mean in the API call to the server to *set* this user's own status is a different question. (One I'd also be glad to have a clear answer to! But that one I feel like I have a more confident guess at than this one -- I'm guessing/hoping that when a parameter is left out there, it simply doesn't change the value the server has for that field.)

Reply:
This is actually API-related, as the `initialData` structures are dependent on the back-end and in another PR these types are moved to `apiTypes` (maybe closed)

Reply:
I did change the `away` type to `away?: true,` which is the precise type. The value never exists as `false`.

Reply:
Thanks for the jsdoc on this type, which answers most of my questions!

> This is actually API-related, as the `initialData` structures are dependent on the back-end

Hmm, it's true that we're using the same type to describe that structure. 

I think that doesn't really change most of what I discussed in my comments above, though. It's fine and good to reuse the same type in our Redux state's type as in describing parts of the server API, where that makes sense... but it doesn't relieve us of our responsibility for defining what the objects in our Redux state tree are supposed to mean. Which the jsdoc you've added largely does. :smile: 

(I have a specific follow-up question or two, which I'll ask as a separate comment-thread.)